loggers/archive.py

```diff
@@ -0,0 +1,70 @@
+from peewee import fn, IntegrityError
+from db import db, archive_db
+from datetime import datetime, timedelta
+import logging
+
+
+def ensure_datetime(timestamp):
+    if isinstance(timestamp, datetime):
+        return timestamp
+    else:
+        return datetime.fromtimestamp(timestamp)
+
+
+@db.atomic()
```

Comment:
so .. this would still corrupt the data if run in parallel (which is the point of using atomic transactions);

worker 1 selects ALL records - [A,B,C,D, ... Z]
slightly later, worker 2 selects ALL records in a transaction.  maybe A is done already, so it gets [C,D,E, ... Z]

so both workers will try to process [ ... Z] and chaos ensues


you want to handle a single record, entirely within a transaction:

worker 1 opens a transaction:
  selects record A
  writes it (successfully)
  deletes it
  closes transaction

worker 2 selects a record - A is completely gone, it processes B.
worker 1 selects C, etc.

this is sort of advanced / extra credit